artemis-protocols/artemis-amqp-protocol/src/main/java/org/apache/activemq/artemis/protocol/amqp/proton/handler/ProtonHandler.java

```diff
@@ -497,6 +498,9 @@ private void dispatch() {
                if (log.isTraceEnabled()) {
                   log.trace("Handling " + ev + " towards " + h);
                }
+               if (AuditLogger.isAnyLoggingEnabled()) {
```

Comment:
Do you need to do this for every event? It seems you only need to this only once.. so I think you could do this at the beggining of the method, before the loop starts.

Reply:
there are no test failures on your branch now.. we just need to resolve the thread local versus thread changes due to asynchronous operations on the broker.